uaclient/util.py

```diff
@@ -53,7 +53,11 @@ def __init__(
         headers: "Optional[Dict[str, str]]" = None,
         url: "Optional[str]" = None,
     ):
-        super().__init__(str(cause))
+        if getattr(cause, "reason", None):
+            cause_error = cause.reason.strerror  # type: ignore
```

Comment:
Why the ignore comment?

Would using cast help, a la: https://github.com/CanonicalLtd/ubuntu-advantage-client/blob/537bedf2990200f975d3aa931b777dcd4a2e2533/uaclient/entitlements/__init__.py#L26

Reply:
(Also, how about `cause_error = getattr(cause, "reason", str(cause))`?